dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/RandomLoadBalance.java

```diff
@@ -69,13 +69,14 @@ protected <T> Invoker<T> doSelect(List<Invoker<T>> invokers, URL url, Invocation
         int totalWeight = 0;
         for (int i = 0; i < length; i++) {
             int weight = getWeight(invokers.get(i), invocation);
+            // judge before accumulation
+            if (sameWeight && totalWeight != weight * i) {
+                sameWeight = false;
+            }
```

Comment:
Is there any different?

Reply:
we don't need a external addition operation

Reply:
we don't need an external addition operation.